test/language/function-code/each-param-has-own-scope.js

```diff
@@ -0,0 +1,21 @@
+// Copyright (C) 2017 Rick Waldron. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-function-definitions-runtime-semantics-iteratorbindinginitialization
+description: A new declarative environment is created, from the originalEnv, for each parameter.
+info: |
+  Runtime Semantics: IteratorBindingInitialization
+
+  FormalParameter : BindingElement
+
+  ...
+  6. Let paramVarEnv be NewDeclarativeEnvironment(originalEnv).
+  ...
+---*/
+
+function f(a = eval("var x = 1; x"), b = x) {}
```

Comment:
maybe you should observe the eval execution by leaking an assignment to a name defined in the outer scope.

```js
var y;
function f(a = eval("var x = 1; y = 42; x"), b = x) {}

assert.throws(ReferenceError, () => {
  f();
});

assert.sameValue(y, 42); 
```

This prevents any false positive.

I also imagine the second parameter with an eval call to assure the `paramVarEnv` is not shared, but I'm not sure how to draft this. Maybe it's too much?

Reply:
> This prevents any false positive.

+ 1 to that change

> I also imagine the second parameter with an eval call to assure the paramVarEnv is not shared, but I'm not sure how to draft this. Maybe it's too much?

Or maybe just another test altogether?